### PR TITLE
Fix macOS extended test (by using printf)

### PR DESCRIPTION
**Pull request by cielavenir**
Because macOS's bash does not accept `echo -n` when launched as /bin/sh

### CODE REVIEW

**Comment by cielavenir** [comment]
To ensure https://github.com/intel/isa-l/pull/278 more, I needed to run extended test on macOS (both x64 and arm64) (https://github.com/cielavenir/isa-l/actions/runs/8130749368/job/22219331109 passed)

**Comment by cielavenir** [comment]
can someone check this?

File: tools/gen_nmake.mk [hunk 1]
```diff
@@ -14,7 +14,7 @@ Makefile.nmake tst.nmake: FORCE
 	@echo ''			>> $@
 	@echo '# This file can be auto-regenerated with $$make -f Makefile.unx Makefile.nmake' >> $@
 	@echo ''			>> $@
-	@echo -n 'objs =' >> $@
+	@/bin/echo -n 'objs =' >> $@
```
Comment (pablodelara) [inline tools/gen_nmake.mk:17]: Why is it needed for only these two lines? because of the -n parameter?

Comment (cielavenir) [reply tools/gen_nmake.mk:17]: yes.

Ii seems like I can use `printf` alternatively, though.

Comment (pablodelara) [reply tools/gen_nmake.mk:17]: Could be a better choice, knowing that this is hardcoding the path.

Comment (cielavenir) [reply tools/gen_nmake.mk]: @pablodelara done.

**Comment by liuqinfei** [comment]
There are three commits in this PR. Actually they can be merged into two commits.

**Comment by cielavenir** [comment]
@liuqinfei rebased, though I'm not sure if I should rebase again after review,

**Comment by pablodelara** [comment]
@cielavenir  Looks ok, but could you change the name of the first commit to be more descriptive? Like "tools: replace echo -n with printf". I can do it for you if you like.

**Comment by cielavenir** [comment]
@pablodelara rebased

**Comment by pablodelara** [comment]
Code is merged, thanks!

**Pull request closed without merging**